openedx/core/djangoapps/user_api/views.py

```diff
@@ -957,6 +959,37 @@ def _apply_third_party_auth_overrides(self, request, form_desc):
                         required=False,
                     )
 
+    def _apply_enterprise_overrides(self, request, form_desc):
+        """
+        Modify the registration form if the user is associated with an enterprise.
+
+        If a user has successfully authenticated with a enterprise SSO,
+        but does not yet have an account with edX, we want to apply enterprise specific overrides to
+        login and registration pages.
+
+        Arguments:
+            request (HttpRequest): The request for the registration form, used
+                to determine if the user has successfully authenticated
+                with a third-party provider.
+
+            form_desc (FormDescription): The registration form description
+
+        """
+        if enterprise_api.enterprise_customer_for_request(request):
+            # Always show the password field for enterprise learners
+            form_desc.override_field_properties(
```

Comment:
I'm concerned that we might decide not to do this for every enterprise customer, especially if their IdP is set to auto submit the registration form, which I know IBM is doing.